src/rust/storage/seg/src/segments/segments.rs

```diff
@@ -111,15 +119,292 @@ impl Segments {
             data,
             flush_at: Instant::recent(),
             evict: Box::new(Eviction::new(segments, evict_policy)),
+            data_file_backed,
+            fields_copied_back: false,
         }
     }
 
+    /// Private function which allocates and initializes the `Segments` by
+    /// taking ownership of the builder.
+    /// `Segments` is restored if the paths are specified, otherwise a new
+    /// `Segments` is created.
+    pub(super) fn from_builder_restore(builder: SegmentsBuilder) -> Self {
```

Comment:
I don't like how much code winds up copied here... consider ways to reduce the duplication